docs/community/contributing.rst

```diff
@@ -0,0 +1,194 @@
+.. highlight:: shell
+
+============
+Contributing
+============
+
+These are the pyfar-wide contributing guidelines. For package specific guidelines please visit the github page for that package, which is linked from *code* buttons on `pyfar.org <https://pyfar.org>`_
+
+Pull Request Guidelines
+-----------------------
+
+Before you submit a pull request, check that it meets these guidelines:
+
+1. The pull request should include tests.
+2. If the pull request adds functionality, the docs should be updated. Put your new functionality into a function with a docstring.
+3. If checks do not pass, have a look at https://app.circleci.com/pipelines/github/pyfar/PACKAGENAME for more information (e.g., use 'pyfar' as the PACKAGENAME).
+
+Function and Class Guidelines
+-----------------------------
+
+Functions and classes should
+
+* have a single clear purpose and a functionality limited to that purpose. Conditional parameters are fine in some cases but are an indicator that a function or class does not have a clear purpose. Conditional parameters are
+
+  - parameters that are obsolete if another parameter is provided
+  - parameters that are necessary only if another parameter is provided
+  - parameters that must have a specific value depending on other parameters
+
+* be split into multiple functions or classes if the functionality is not well limited.
+* contain documentation for all input and output parameters.
+* contain examples in the documentation if they are non-trivial to use.
+* contain comments in the code that explain decisions and parts that are not trivial to read from the code. As a rule of thumb, too many comments are better than too few comments.
```

Comment:
```suggestion
* contain comments in the code that explain decisions and parts that are not trivial to read from the code. As a rule of thumb, too many comments are better than too few comments. But if every line must be described, the code is not well written.
```
This might just be my own opinion. But to me, good code is already easy to read without comments. So feel free to skip this comment.